lib/String.ts

```diff
@@ -36,5 +36,6 @@ export const extractNumbers = (str: string) => {
  * An interface reprsenting types that have a callable `toString` method.
  */
 export interface ToStringable {
-  toString(): string
+  toString(): string;
+  toJSON?: () => string;
```

Comment:
separate tojson out

Reply:
I still wouldn't include it as an optional here, as this interface is only meant to represent `toString`. In the case of the URL functions you could easily have a type like:
```ts
export interface ToStringable {
  toString(): string
}

export interface ToJSONable {
  toJSON(): string
}

export type URLConvertible = ToStringable | ToJSONable
```

However, I think in this case for URLs, instead of relying on `toJSON` I would rather have something like:
```ts
export interface URLQueryParameterConstructable {
  toURLQueryParameter(): string
}
```
URL query parameters have a very specific format, and that format is not JSON. If you want to omit the `@` sign from the `UserHandle` in the URL, you can then conform it to that interface where it returns the raw string without the `@`.

Then `URLParameters` can look like:
```ts
export type URLParameters = {
  [key: string]: ToStringable | URLQueryParameterConstructable | undefined
}
```
In `queryToSearchParams` or other functions, you can easily check to see if it has the `toURLQueryParameter` method. If the object doesn't have such a method then fallback to `toString`.